libarchive/archive_write_set_format_gnutar.c

```diff
@@ -523,7 +523,11 @@ archive_write_gnutar_header(struct archive_write *a,
 			goto exit_write_header;
 	}
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
+	if (archive_entry_hardlink_w(entry) != NULL) {
+#else
 	if (archive_entry_hardlink(entry) != NULL) {
```

Comment:
A cleaner fix would be to introduce a new `archive_entry_hardlink_is_set()` test.

Reply:
I can add that